src/composer/src/cossystem.rs

```diff
@@ -28,13 +28,20 @@ pub struct InterfaceVariant {
     pub variant: Option<String>,
 }
 
+#[derive(Debug, Deserialize,Clone)]
+pub struct ConstantVal {
+    pub variable: String,
+    pub value: String,
+}
+
 #[derive(Debug, Deserialize)]
 pub struct TomlComponent {
     name: String,
     img: String,
     baseaddr: Option<String>,
     deps: Option<Vec<Dep>>,
     params: Option<Vec<Parameters>>,
+    constants: Option<Vec<ConstantVal>>, // field for constants
```

Comment:
Not sure if the comment is adding anything here. If you want it to be useful, add more context that the reader cannot get from the names. "Set of configurable constants used to specialize a component's compilation." for example.